mos_tests/neutron/python_tests/test_restarts.py

```diff
@@ -71,21 +74,37 @@ def prepare_openstack(self, setup):
         self.check_vm_connectivity()
 
         # Find a primary contrloller
-        for node in self.env.get_all_nodes():
-            # TBD this is a temp solution.
-            # Te be replaced by 'heira role' in next commit
-            if node.data['name'] == 'slave-01_controller':
-                self.primary_node = DevopsClient.get_devops_node(
-                                        node.data['name'].split('_')[0])
-                self.primary_host = node.data['fqdn']
-                break
+        self.primary_node = DevopsClient.get_devops_node(
+            self.env.primary_controller.data['name'].split('_')[0])
+        self.primary_host = self.env.primary_controller.data['fqdn']
 
         # make a list of all l3 agent ids
         self.l3_agent_ids = [agt['id'] for agt in
                                 self.os_conn.neutron.list_agents(
                                     binary='neutron-l3-agent')['agents']]
 
+        self.dhcp_agent_ids = [agt['id'] for agt in
+                                self.os_conn.neutron.list_agents(
+                                    binary='neutron-dhcp-agent')['agents']]
+
     def test_shutdown_primary_controller_with_l3_agt(self):
+        """Id of the test is C57896
+            8. Check on what agents is router1:
```

Comment:
The numeration is stared in the fixture in the same class. Since the first actions are common for all tests.